vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/main/java/com/vaadin/flow/component/grid/it/GridDelegateFocusToComponentPage.java

```diff
@@ -45,18 +45,12 @@ public GridDelegateFocusToComponentPage() {
     }
 
     private Div buildComplexComponent(String item) {
-        TextField hiddenTextField = new TextField();
-        hiddenTextField.setValue("hidden");
-        hiddenTextField.getElement().setAttribute("hidden", true);
-        TextField disabledTextField = new TextField();
-        disabledTextField.setValue("disabled");
-        disabledTextField.setEnabled(false);
```

Comment:
When entering from cell navigation mode to interaction mode (by pressing enter), `<vaadin-grid>` Web Component [tries to automatically focus](https://github.com/vaadin/web-components/blob/ed28a0f1f3b2a40ed26c25ffbfe915a1487b9cc3/packages/grid/src/vaadin-grid-keyboard-navigation-mixin.js#L610) the element with "focus-target" attribute or the first focusable element in the cell.

The `<flow-component-renderer>` had some [custom logic](https://github.com/vaadin/flow-components/blob/99682fa0fbe06134cca77c9b7ed7da554f701266/vaadin-renderer-flow-parent/vaadin-renderer-flow/src/main/resources/META-INF/resources/frontend/flow-component-renderer.js#L110) to automatically mark an element as "focus-target" _when the node was appended_. This custom logic had additional checks to ignore disabled and hidden elements, which is currently not properly covered by the Web Component's [`isFocusable` check](https://github.com/vaadin/web-components/blob/ed28a0f1f3b2a40ed26c25ffbfe915a1487b9cc3/packages/grid/src/vaadin-grid-active-item-mixin.js#L106).

Since not using `<flow-component-renderer>` falls back to using the Web Component focusing logic, I needed to disabled the checks in this test. The `isFocusable` check in `<vaadin-grid>` should be fixed to properly ignore hidden and disabled nodes instead.

Reply:
Submitted a [PR](https://github.com/vaadin/web-components/pull/5263) to fix the issue in `<vaadin-grid>`